pkg/utils/perm_windows.go

```diff
@@ -0,0 +1,15 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build windows
+// +build windows
+
+package utils
+
+// HasStrictExecPerms ensures that the path is executable by the owner and that the owner of the file
+// is the same as the UID or root.
+func HasStrictExecPerms(path string, uid int) error {
+	// TODO: Need to add check on Windows to ensure that the ACL are correct for the binary before execution.
```

Comment:
Is this coming in this PR or do we need a follow up issue to do this?

Reply:
I will file a follow up issue so we can add it in a follow up issue. The windows code is very windows specific and will require some time with me actually on Windows to develop it.

Reply:
Added this as a TODO on the V2 tracking issue with a link to this comment